src/Database/Database.php

```diff
@@ -58,7 +62,7 @@ public function __construct(IConfigCache $configCache, Profiler $profiler, Logge
 		$this->logger        = $logger;
 		$this->db_serveraddr = $serveraddr;
 		$this->db_user       = $user;
-		$this->db_pass       = $pass;
+		$this->db_pass       = new HiddenString($pass);
```

Comment:
This is a start, but the ideal would be to pass an already constructed HiddenString in the constructor so that the plaintext password can't show as a parameter of the constructor call either.

Reply:
Ha .. Didn't thought about that possible security-issue .. thanks!

Reply:
I owe almost everything I know about PHP security to @paragonie and @paragonie-scott. Praise be them.